lib/puppet/configurer.rb

```diff
@@ -425,6 +432,18 @@ def run_internal(options)
   end
   private :run_internal
 
+  def valid_server_environment?
+    session = Puppet.lookup(:http_session)
+    begin
+      fs = session.route_to(:fileserver)
+      response = fs.get_file_metadatas(path: URI(Puppet[:pluginsource]).path, recurse: :false, environment: @environment)
+      return response[0].success?
```

Comment:
All of the http services raise if the request is not successful, so you could simplify to:

```ruby
begin
  fs = session.route_to(:fileserver)
  fs.get_file_metadatas(path: URI(Puppet[:pluginsource]).path, recurse: :false, environment: @environment)
  true
rescue ...
```